### PR TITLE
Replace openssl with boringssl as the official ssl provider

### PR DESCRIPTION
**Pull request by RomanDzhabarov**
cc: @lyft/network-team @PiotrSikora

### CODE REVIEW

File: ci/build_container/build_container.sh [hunk 1]
```diff
@@ -43,6 +35,18 @@ make install
 cd ..
 rm -fr libevent*
 
+# boring ssl
```
Comment (PiotrSikora) [inline ci/build_container/build_container.sh]: Nit: BoringSSL.

File: ci/build_container/build_container.sh [hunk 2]
```diff
@@ -43,6 +35,18 @@ make install
...
+git clone https://boringssl.googlesource.com/boringssl
+cd boringssl
+git reset --hard 78684e5b222645828ca302e56b40b9daff2b2d27
+cmake -DCMAKE_INSTALL_PREFIX:PATH=$THIRDPARTY_BUILD .
```
Comment (PiotrSikora) [inline ci/build_container/build_container.sh]: BoringSSL doesn't provide `make install` target, so `-DCMAKE_INSTALL_PREFIX:PATH=$THIRDPARTY_BUILD` is useless and can be removed.

**Comment by mattklein123** [comment]
Does this work without any other code changes? Or do we also need other commits to change paths?

**Comment by PiotrSikora** [comment]
Commit message could use a bit more text ;)

**Comment by mattklein123** [comment]
If there are no other code changes required, can you also update docs in this commit about third party dependencies. (You should mention in docs that we build against boringssl but that openssl should still work).

File: ci/build_container/build_container.sh (hunk 2 above)
Comment (RomanDzhabarov) [inline ci/build_container/build_container.sh]: right, i was playing around it and forgot to remove...

**Comment by RomanDzhabarov** [comment]
new image is pushed to docker hub

**Pull request merged**